contracts/sweep/SweepERC20.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./BaseTokenSweep.sol";
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+/**
+ * @title Adds the ability to sweep a token to a beneficiary address
+ *
+ * @dev
+ */
+abstract contract SweepERC20 is BaseTokenSweep {
+    /**
+     * @notice Sweep the erc20 tokens to the beneficiary address
+     *
+     * @dev Needs access control implemented in the inheriting contract
+     **/
+    function sweepERC20Tokens(IERC20Upgradeable token, uint256 amount)
```

Comment:
suggestion: deal with primitives in external / public functions.

The EVM is not typed. `token` is not guaranteed to be of type `IERC20Upgradeable`, rather it is cast to that type from whatever it may actually be.

There is ERC165, which was/is an attempt to navigate this issue, but in short, you can't trust the type of the input.
Instead you'll notice `address` is used, with it being cast  in the line containing transfer. The usual motivation (or at least initially) was to later on inject another `require` using the ERC165 standard to query the address given.

Reply:
Interesting! Changed to primitive. Was aware the EVM doesnt care - Had been previously by auditors to have as much as possible typed.